backend/src/main/java/uk/ac/cam/cl/kilo/Server.java

```diff
@@ -1,83 +1,55 @@
 package uk.ac.cam.cl.kilo;
 
 import java.io.*;
-import javax.servlet.*;
-import javax.servlet.http.*;
-import java.util.*;
+import static spark.Spark.*;
+package uk.ac.cam.cl.kilo.database;
 
 public class Server extends HttpServlet {
- 
-    @Override
-    //Handle GET request
-    public void service(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-        if (request.getMethod().equals("POST")) {
-            doPost(request, response);
+  public static void main(String args[]){
 
-        } else if (request.getMethod().equals("GET")) {
-            doGet(request, response);
-
-        }
-        //otherwise consider redirect from FB
-        // put info from FB into DB
-        request.getRequestURL();
-        return;
-    }
-
-    @Override
-    //Handle GET request
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-       return;
-    }
-
-    // Handle POST request
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-        // Set the response content type
-        response.setContentType("text; charset=UTF-8");
-        PrintWriter out = response.getWriter();
-
-        try {
-            String state = request.getParameter("state");
-            if(state == null || htmlFilter(state).isEmpty()){
+    get("/facebook/register/stage1", (request, response) -> {
+            String state = request.queryParamOrDefault("state","");
```

Comment:
We don't want to accept state as a parameter here, we want to generate the state. I added a function for doing this to `Server.java`, but you haven't pulled that version. Try:

```
git remote add upstream https://github.com/groupkilo/heroes_in_conference.git
git pull upstream master
```